.github/workflows/external.yml

```
@@ -21,20 +23,47 @@ jobs:
           python-version: 3.9
           cache: "pip"
 
-      - name: Install dependencies
+      - name: Build sdist
         run: |
-          pip install --upgrade pip
-          pip install -r requirements.txt
-          pip install -r requirements-dev.txt
+          python -m pip install --upgrade pip
+          python -m pip install -U build pip setuptools
+          python -m pip install -U -r requirements.txt
+          python -m build --sdist
+
+      - name: Delete source directory
+        shell: bash
+        run: |
+          rm -rf $MODULE_NAME
+
+      - name: Uninstall all packages
+        run: |
+          python -m pip freeze > installed.txt
+          python -m pip uninstall -y -r installed.txt
+
+      - name: Install from sdist
+        shell: bash
+        run: |
+          SDIST=$(python -c "import os;print(os.listdir('./dist')[-1])" 2>&1)
+          python -m pip install dist/$SDIST
+
+      - name: Test import
+        shell: bash
+        run: |
+          python -c "import $MODULE_NAME" -Werror
+
+      - name: Install test requirements
+        run: |
+          python -m pip install -U -r requirements.txt
+          python -m pip install -U -r requirements-dev.txt
           python -m spacy download en_core_web_md
```

There is some chance that `spacy download` (without `--no-deps`) would clobber some of the other requirements, so I think it's better to prioritize the local requirements. It shouldn't, but it could be hard to catch here if it does.

```suggestion
          python -m spacy download en_core_web_md
          python -m pip install -U -r requirements.txt -r requirements-dev.txt
```